calcom/cal.com
# feat: add Adyen Payment App

## What does this PR do?

- Fixes #16860 (GitHub issue number)
- Lets users provide their credentials to process on cal.com. 
- Fixes CAL-16860

### Loom videos
- [App Store Listing](https://www.loom.com/share/bc54acf660fb4699a1799ca452717a43?sid=6fbd89dc-5807-4083-b5d7-cbe71d1a5cec)
- [Setup Screen](https://www.loom.com/share/d795bcac7b384361a82f69b378f01592?sid=22bcefa3-a4d9-4d5b-973d-bdc3fe9b99c3)
- [Payment Confirmation](https://www.loom.com/share/6124bbcb95f44e76be927f1289c13471?sid=9bac48b2-c3a9-43cf-b36d-88585aa7eadd)

### <ins>**Important**<ins>
It should be noted that Adyen requires an agent to confirm and approve the integration implementation before providing the live API keys. 

I've assumed part of integration is to leave the onus of procuring the credentials/accounts to the user. However, while its not explicity required, we may need to contact an Adyen representative regardless.

Also, Adyen requires a reachable hostname to run a mock payment, so local development without tools like ngrok will not yield entire results.

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?

Identical payment app tests are added for the playwright test suite. These include payment adjustments. These can be run via `yarn test-e2e:app-store`
- Are there environment variables that should be set?
For live environments, ADYEN_LIVE_ENVIRONMENT must be set to true.
- What are the minimal test data to have?
Asides from manual e2e testing and playwright tests without API calls, none.
- What is expected (happy path) to have (input and output)?
A given user can switch to their Adyen setup, where they can then specify their payments via their Adyen account. 
- Any other important info that could help to test that PR

## Checklist

/claim #16860
